tests/parametric/test_sampling_span_tags.py

```diff
@@ -0,0 +1,118 @@
+import json
+
+import pytest
+from utils import bug, scenarios
+from utils.parametric.spec.trace import MANUAL_DROP_KEY  # noqa
+from utils.parametric.spec.trace import MANUAL_KEEP_KEY  # noqa
+from utils.parametric.spec.trace import SAMPLING_AGENT_PRIORITY_RATE  # noqa
+from utils.parametric.spec.trace import SAMPLING_DECISION_MAKER_KEY  # noqa
+from utils.parametric.spec.trace import SAMPLING_PRIORITY_KEY  # noqa
+from utils.parametric.spec.trace import SAMPLING_RULE_PRIORITY_RATE  # noqa
+from utils.parametric.spec.trace import Span  # noqa
+from utils.parametric.spec.trace import find_span_in_traces  # noqa
+
+
+def _get_parent_and_child_span(test_agent, test_library):
+    with test_library:
+        with test_library.start_span(name="parent", service="webserver") as parent_span:
+            with test_library.start_span(
+                name="child", service="webserver", parent_id=parent_span.span_id
+            ) as child_span:
+                pass
+
+    traces = test_agent.wait_for_num_spans(2, clear=True)
+
+    parent_span = find_span_in_traces(traces, Span(name="parent", service="webserver"))
+    child_span = find_span_in_traces(traces, Span(name="child", service="webserver"))
+    return parent_span, child_span
+
+
+def _assert_sampling_tags(parent_span, child_span, child_dm, parent_dm, parent_priority, parent_rate):
+    if child_dm is not None or "meta" in child_span:
+        assert child_span["meta"].get(SAMPLING_DECISION_MAKER_KEY) == child_dm
+    if parent_dm is not None or "meta" in parent_span:
+        assert parent_span["meta"].get(SAMPLING_DECISION_MAKER_KEY) == parent_dm
+    assert parent_span["metrics"].get(SAMPLING_PRIORITY_KEY) == parent_priority
+    assert parent_span["metrics"].get(SAMPLING_RULE_PRIORITY_RATE) == parent_rate
+
+
+@scenarios.parametric
+class Test_Sampling_Span_Tags:
+    @pytest.mark.parametrize("library_env", [{"DD_TRACE_SAMPLE_RATE": 1}])
+    @bug(library="python", reason="Python sets dm tag on child span")
+    @bug(library="python_http", reason="Python sets dm tag on child span")
```

Comment:
@cbeauchesne Should we be only testing with http endpoints? Are we planing to deprecate grpc endpoints in parametric tests.

Reply:
@Kyle-Verhoog what is your opinion for that ?

Reply:
My instinct would be the other way around. Interested to hear what Kyle thinks.

Reply:
I tried to push for us to change over to http since not all the languages can support grpc (php) but the message didn't get out there and now we have everyone (except php) using grpc 😕.

@pawelchcki since the shared library interface isn't _huge_ yet, can we prioritize the change over to http?

@dgoffredo @mabdinur IMO we should be testing with both python grpc and python http in the interim if it works. If it doesn't then I'd say right now to prioritize GRPC over http until R&P makes a decision.

Reply:
I'm also in favor of http, but wanted to have your opinion first.

We'll talk about this with @pawelchcki